README.md

```diff
@@ -0,0 +1,166 @@
+
+Now that we've built our "Hello World" server, let's start to build out our messaging apparatus. The ability to add messages seems like a good place to start, no? Then we can work on our retrieval pathways.
+
+So what do we need to do? First, of all we need place to store the messages. Let's just store our messages in memory to keep things simple. This will mean that the messages we create won't persist if we shut down the server, but that's okay for our purposes. To achieve this, let's define a variable `messages` that will start as an empty list or Array. You can put this somewhere near the top of your `server.js` file:
+
+`let messages = [];`
+
+Note, that in the line above we are using the new ES6 variable assignment keyword `let`. You will need to put the compiler into strict mode for this to work, otherwise you'll likely get an error when you run the code. (To put the compiler into strict mode, just add `"use strict";` to the top of the file.)
+
+Next let's define a new route for submitting messages. In order to send data to the server, we want to use a different HTTP request "method." For the Hello World server we used a standard GET request, but for sending messages we'll use the POST method. A code stub like this will get us started:
+
+```javascript
+route.post('/message', (request, response) => {
+  // Save the message and send the message id back to the client.
+});
+```
+
+At this point, our server can catch a POST request to the path `/message`. Great! But how will we get our message data into the system? Now, normally you'd probably have a web frontend that submits the data through the browser. As mentioned above, we'll simulate this by using curl. The following command will submit a message to our new path:
+
+```bash
+curl -X POST http://localhost:3000/message -d '{"id":1,"message":"Our first message."}'
+```
+
+Okay, so this is how we'll submit data, but how will we read in our route's logic? Here's where we will use the `request` object that is passed to our callback function as an argument. However, although this object contains a lot of information about the request being made to the server, by default it does not contain the data that we've sent across in our curl request in a way that is easily accessed. To make things a bit easier, we'll use another Express module called [body-parser](https://github.com/expressjs/body-parser). As its name suggests, this module parses data that the server receives from the client in the request body. It then attaches this parsed data to the request object's `body` parameter.
+
+The body-parser is also a special kind of module known as "middleware." You'll be using middleware frequently when programming in Node. As its name suggests, middleware is software that runs in between; that is, it is code that, in this case, is executed after a request is sent to the server, and before the request is processed by the server itself.
```

Comment:
You mentioned this in our sync-up, and I think you're right: maybe we can clarify this by changing the last sentence to 

"As its name suggests, middleware is software that runs _in between_; that is, it is code that, in this case, is executed after a request is sent to the server, and before the server sends its response."